tracer/test/test-applications/integrations/Samples.InstrumentedTests/Vulnerabilities/ReflectionInjection/ActivatorTests.cs

```diff
@@ -0,0 +1,245 @@
+    protected string notTaintedType;
+
+    protected string assemblyTainted;
+    protected string typeTainted;
+
+    public ActivatorTests()
+    {
+        assembly = Assembly.GetExecutingAssembly().GetName().Name;
+        assemblyFile = Assembly.GetExecutingAssembly().Location;
+        notTaintedType = GetType().BaseType!.FullName;
+
+        typeTainted = typeof(ActivatorTestType).FullName;
+        assemblyTainted = typeof(string).Assembly.GetName().Name;
+
+        AddTainted(assemblyTainted);
+        AddTainted(typeTainted);
+    }
+
+#if !NETCOREAPP2_1
+    [Fact]
+    public void GivenAnActivator_WhenCallingCreateInstance2TaintedType_VulnerabilityIsReported()
+    {
+        Activator.CreateInstance(assembly, typeTainted);
+        AssertVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnActivator_WhenCallingCreateInstance2TaintedAssembly_VulnerabilityIsReported()
+    {
+        var type = typeof(DateTime).FullName;
+        Activator.CreateInstance(assemblyTainted, type);
+        AssertVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnActivator_WhenCallingCreateInstanceTaintedAssembly_VulnerabilityIsReported()
+    {
+        AddTainted(assembly);
+        Activator.CreateInstance(assembly, notTaintedType);
+        AssertVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnActivator_WhenCallingCreateInstanceTaintedTypeAssembly_VulnerabilityIsReported()
+    {
+        AddTainted(assembly);
+        Activator.CreateInstance(assembly, typeTainted);
+        AssertVulnerable(vulnerabilities: 2);
+    }
+
+    [Fact]
+    public void GivenAnActivator_WhenCallingCreateInstanceNotTainted_VulnerabilityIsNotReported()
+    {
+        Activator.CreateInstance(assembly, notTaintedType);
+        AssertNotVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnActivator_WhenCallingCreateInstanceTaintedTypeNullParameters_VulnerabilityIsReported()
+    {
+        Activator.CreateInstance(assembly, typeTainted, activationAttributes: null);
+        AssertVulnerable();
+    }
+
+    [Fact]
+    public void GivenAnActivator_WhenCallingCreateInstanceTaintedTypeParameters_VulnerabilityIsReported()
+    {
+        Activator.CreateInstance(assembly, typeTainted, new object[] { });
+        AssertVulnerable();
+    }
+#endif
+#if NETFRAMEWORK
+    [Fact]
+    public void GivenAnActivator_WhenCallingCreateInstanceTaintedTypeFlagsCulture_VulnerabilityIsReported()
+    {
+        Activator.CreateInstance(assembly, typeTainted, ignoreCase: true, BindingFlags.Default, binder: null, args: null, CultureInfo.InvariantCulture, activationAttributes: null);
+        AssertVulnerable();
+    }
+#endif
+#if NETFRAMEWORK
+    [Fact]
+    public void GivenAnActivator_WhenCallingCreateInstanceDomainTaintedType_VulnerabilityIsReported()
+    {
+        Activator.CreateInstance(AppDomain.CurrentDomain, assembly, typeTainted);
+        AssertVulnerable();
+    }
+#endif
+#if !NET35 && !NETCOREAPP1_0_OR_GREATER
```

Comment:
Those directives are no longer used in DD and are Hdiv inheritance. I would remove all NET35 references. Also, in this project, !NETCOREAPP1_0_OR_GREATER would be equivalent to NETFRAMEWORK.

Reply:
Isn't these still valid as per the doc? https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives
But ok I will change them 👍

Reply:
We don't compile in Net35 so, while it's a valid directive, it is useless in our projects .